### ISSUE TITLE
compilation

### ISSUE THREAD

**Issue by ajneu** [NoAvailableAlias/signal-slot-benchmarks#1]
Hi,

on what platform are you compiling/running this?

Under gcc not even a basic compilation to object-file of the toplevel `benchmark.cpp` works:

``` bash
c++ -std=c++11 -c benchmark.cpp
```

(seems to me that the different headers are interfering with one another. perhaps one needs to "ban" every implementation to it's own cpp... or rework those headers somehow... )
.
.
.

PS: Isn't the jeffomatic jl_signal... result of the combined test, all that counts??? (The combined test includes all other tests anyway, right?)  Why would one want to weight the construction test so highly???

.
.
.
PS: what kind of connections do you test? static functions, Class members, functors, lambdas... ??? (I have not checked the code)

**Comment by NoAvailableAlias** [comment]
The benchmarks were compiled and run with Visual Studio 2015 community under release build settings.

I haven't had a linux box to test on gcc since a couple years (been getting by gcc testing for nano-signal-slot by testing on coliru) so I'm not 100% positive of correctness. Other than making sure you have boost installed, linking the boost.signals lib, all that should be required is compile all .cpp in single gcc command.

The benchmark scores aren't weighted at all and is up to the viewer to decide how to weight them. (this means i don't have to create some sort of weighting scheme someone will find fault with). I do agree that construction timing shouldn't be worth much and you are right that the combined results are more important. I just needed a quick way to sort results and horizontal add was simplest.

Connections are to class instance methods.

**Comment by NoAvailableAlias** [comment]
This issue will be closed once the rework branch is merged into master.

This merge will add a complete Visual Studio 17 solution as well as cmake support.
